source-mysql/.snapshots/TestDatetimeNormalization-replication

```diff
@@ -0,0 +1,12 @@
+# ================================
+# Collection "acmeCo/test/test_datetimenormalization_24528211": 4 Documents
+# ================================
+{"_meta":{"op":"c","source":{"ts_ms":1111111111111,"schema":"test","table":"DatetimeNormalization_24528211","cursor":"binlog.000123:56789:123","txid":"11111111-1111-1111-1111-111111111111:111"}},"id":200,"x":"1991-08-31T17:34:57Z"}
+{"_meta":{"op":"c","source":{"ts_ms":1111111111111,"schema":"test","table":"DatetimeNormalization_24528211","cursor":"binlog.000123:56789:123","txid":"11111111-1111-1111-1111-111111111111:111"}},"id":201,"x":"0001-01-01T00:00:00Z"}
+{"_meta":{"op":"c","source":{"ts_ms":1111111111111,"schema":"test","table":"DatetimeNormalization_24528211","cursor":"binlog.000123:56789:123","txid":"11111111-1111-1111-1111-111111111111:111"}},"id":202,"x":"2022-11-30T06:00:00Z"}
```

Comment:
I'm understanding that this corresponds to the input `"2023-00-00 00:00:00"` - but I'm not understanding how the `2023` year part gets turned into a `2022`? It seems like we are only adding to the input. Is this just something weird happening with MySQL that I shouldn't worry too much about?

Reply:
Yeah, that's the inconsistency I referred to. I have absolutely no idea why we're seeing `2023-00-00` turned into `2022-11-30` (note that this is completely insane, it should _at least_ be `2022-12-31` if things are getting normalized backwards like that) but I've confirmed that it's nothing we're doing -- that's what the literal replication events coming from the database appear to say (though I have not yet ruled out a client library bug in how the binary representation of these datetimes is parsed, so it still might be something we could fix in theory).

Reply:
I'm not entirely sure (still haven't finished chasing down the relevant parsing code), but _I think_ that the MySQL replication protocol might actually be representing datetime values as a 4-byte Unix timestamp, which would kind of halfway explain why things get normalized oddly there and make it Not Our Problem, in my opinion.